beater/server.go

```diff
@@ -101,13 +101,13 @@ func createHandler(p processor.Processor, config Config, publish successCallback
 
 		// Limit size of request to prevent for example zip bombs
 		limitedReader := io.LimitReader(reader, config.MaxUnzippedSize)
-		err = p.Validate(limitedReader)
+		payload, err := p.Validate(limitedReader)
 		if err != nil {
 			sendError(w, r, 400, fmt.Sprintf("Data Validation error: %s", err), true)
 			return
 		}
 
-		list := p.Transform()
+		list := payload.Transform()
```

Comment:
I think we currently mix `Validation` and `Transformation` together, which is more appearant now than it was before. 
How about clearly separating those two steps, e.g.
```
var buf bytes.Buffer
dataReader := io.TeeReader(reader, &buf)
err := processor.Validate(dataReader)
//do some error check
processor.Transform(buf.Bytes)
```

I have not done a test implementation, but I think it should also be easier to keep the internal logic private then.